tests/ui/traits/super-subtraits/supertrait-in-subtrait-impl.rs

```diff
@@ -0,0 +1,17 @@
+#![feature(rustc_attrs)]
+
+trait Sup {
+    type A;
+}
+
+#[rustc_supertrait_in_subtrait_impl]
+trait Sub: Sup {}
+
+struct S;
+impl Sub for S {
+    //~^ ERROR: the trait bound `S: Sup` is not satisfied
+    type A = ();
+    //~^ ERROR: the trait bound `S: Sup` is not satisfied
```

Comment:
This is not the desired end game, yet. This is rather a demonstration that this patch allows name resolution to associate `A` with that from `trait Sup` correctly.